Doc/library/weakref.rst

```diff
@@ -167,6 +167,21 @@ See :ref:`__slots__ documentation <slots>` for details.
    application without adding attributes to those objects.  This can be especially
    useful with objects that override attribute accesses.
 
+   Note that when a key with equal value to an existing key (but not equal identity)
+   is inserted into the dictionary, it replaces the value but does not replace the
+   existing key. Due to this, when the reference to the original key is deleted, it
+   also deletes the entry in the dictionary::
+
+      >>> class T(str):
+      ...     pass
+      ...
+      >>> k1, k2 = T(), T()
+      >>> d = weakref.WeakKeyDictionary()
+      >>> d[k1] = 1   # d = {k1: 1}
```

Comment:
Maybe this could instead be 

```
# d = {id(k1): 1}
# d = {id(k1): 2}
```

Reply:
That is not true I think. The key of `d` is `k1`, not `id(k1)`

Reply:
Ah yeah you're right, nevermind me. I was thinking of a way to emphasize that the ids are the same but this isn't quite right